smack-experimental/src/main/java/org/jivesoftware/smackx/spoiler/element/SpoilerElement.java

```diff
@@ -0,0 +1,101 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.spoiler.element;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+import org.jivesoftware.smackx.spoiler.SpoilerManager;
+
+public class SpoilerElement implements ExtensionElement {
+
+
+    public static final String ELEMENT = "spoiler";
+
+    private final String hint;
+    private final String language;
+
+    /**
+     * Create a new empty SpoilerElement.
+     */
+    public SpoilerElement() {
+        this(null);
+    }
+
+    /**
+     * Create a new SpoilerElement with a hint about the content.
+     *
+     * @param hint hint about the content.
+     */
+    public SpoilerElement(String hint) {
+        this(null, hint);
+    }
+
+    /**
+     * Create a new SpoilerElement with a hint about a content and a language attribute.
+     *
+     * @param language language of the hint.
+     * @param hint hint about the content.
+     */
+    public SpoilerElement(String language, String hint) {
+        this.language = language;
+        this.hint = hint;
```

Comment:
The constructor should check that hint is not null or empty in case language is not null or empty.